specification/network/resource-manager/Microsoft.Network/stable/2024-01-01/examples/WafPolicyCreateOrUpdate.json

```diff
@@ -365,6 +418,59 @@
                 "matchVariable": "RequestArgValues",
                 "selectorMatchOperator": "StartsWith",
                 "selector": "test"
+              },
+              {
+                "exclude": "Request",
+                "matchVariable": "RequestURI",
```

Comment:
selector wont be present here? What is the difference between selector and values?

Reply:
Right, missed that earlier. Selector has to be removed from required. Both Selector and SelectorMatchOperators will be optional going forward.

Values field is used for exclusion only when Exclude="Request".  For example, if we want to add exclusion on a request header (Test-Key=Test-Value) based on it's key and value contents, we would add:

            {
              "exclude": "Request",
              "matchVariable": "RequestHeader",
              "selectorMatchOperator": "Equals",
              "selector": "Test-Key",
              "valueMatchOperator": "Equals",
              "values": "Test-Value",
            }

Reply:
Ok.. in that case, are you throwing BadRequest error if exclude is set to "MatchVariable" or not present and valueMatchOperator, values fields are set? One other way of handling such patterns is to use polymorphic objects.

Reply:
Yes we have that validations in the RP to throw a bad request error. I explored polymorphic objects, but we have a lot of complex validations required for our use-case, so will keep those checks in RP.